openssl/src/x509/store.rs

```diff
@@ -81,11 +88,53 @@ impl X509StoreBuilder {
 
 impl X509StoreBuilderRef {
     /// Adds a certificate to the certificate store.
-    // FIXME should take an &X509Ref
-    pub fn add_cert(&mut self, cert: X509) -> Result<(), ErrorStack> {
+    pub fn add_cert(&mut self, cert: &X509Ref) -> Result<(), ErrorStack> {
         unsafe { cvt(ffi::X509_STORE_add_cert(self.as_ptr(), cert.as_ptr())).map(|_| ()) }
     }
 
+    /// Sets the maximum verification depth, or the maximum number of intermediate CA certificates that can appear in a chain.
+    pub fn set_depth(&mut self, depth: c_int) -> Result<(), ErrorStack> {
+        unsafe { cvt(ffi::X509_STORE_set_depth(self.as_ptr(), depth)).map(|_| ()) }
+    }
+
+    cfg_if! {
+        if #[cfg(any(ossl102, libressl261))] {
+             /// Sets the verification flags used to verify a certificate against a chain.
+            pub fn set_flags(&mut self, flags: X509VerifyFlags) -> Result<(), ErrorStack> {
+                unsafe { cvt(ffi::X509_STORE_set_flags(self.as_ptr(), flags.bits())).map(|_| ()) }
+            }
+        } else {
+            /// Sets the verification flags used to verify a certificate against a chain.
+            pub fn set_flags(&mut self, flags: c_ulong) -> Result<(), ErrorStack> {
+                unsafe { cvt(ffi::X509_STORE_set_flags(self.as_ptr(), flags)).map(|_| ()) }
+            }
+        }
+    }
+
+    /// Sets the purpose used to verify the certificate chain.
+    pub fn set_purpose(&mut self, purpose: c_int) -> Result<(), ErrorStack> {
```

Comment:
These should not take a raw integer.

Reply:
Will add the trust/purpose values as defined in https://github.com/openssl/openssl/blob/OpenSSL_1_1_1a/include/openssl/x509.h#L116 and https://github.com/openssl/openssl/blob/OpenSSL_1_1_1a/include/openssl/x509v3.h#L412